### ISSUE TITLE
Drop support for Ruby 3.0

### ISSUE THREAD

**Issue by sarahraqueld** [thoughtbot/factory_bot#1715]
[Ruby 3.0 reached EOL on April 2024](https://www.ruby-lang.org/en/downloads/branches/), but we're still running that version on CI:

![Screenshot 2024-11-06 at 11 10 44](https://github.com/user-attachments/assets/b3449fbe-fef1-49ca-a648-8fc6032b22fb)

Editing the workflow build file to remove 3.0 should be enough to close this issue: https://github.com/thoughtbot/factory_bot/blob/main/.github/workflows/build.yml

**Comment by CodeMeister** [comment]
If Ruby 3.0 is working with Rails 7.0 and 7.1 then it looks like it's the gem resolution with Rails 6.1 is the issue.

I'm hesitant for 3.0 to be removed as there are still smaller companies using this version. (yes, they should really upgrade, but many don't have the budget available)

A simpler resolution to stop the automated tests failing would be to exclude Rails 6.1 when testing Ruby 3.0.

However, if FactoryBot now needs gems that don't run on 3.0 then I would suggest bumping the major version number, so Ruby 3.0 developers will know not jump versions.

**Comment by neilvcarvalho** [comment]
Yep, I think we can support 3.0 (in general, one or two EOL versions?) until it becomes too hard to do so. The Ruby 3.0 / 6.1 issue was solved, and was happening because we were wrongly saving the Appraisals generated Gemfile.lock files, which led to impossible gem resolutions for this Ruby/Rails combination (that is valid, by the way).

**Comment by neilvcarvalho** [comment]
Closing as the build issue has been solved.
